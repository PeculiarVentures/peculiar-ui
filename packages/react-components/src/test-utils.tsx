import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
