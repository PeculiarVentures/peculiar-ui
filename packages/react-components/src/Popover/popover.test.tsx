import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { Popover } from './index';

describe('<Popover />', () => {
  it('should render with default styles', () => {
    const {
      baseElement,
    } = render(
      <Popover
        open={false}
        anchorEl={document.createElement('div')}
      >
        <div>Inside</div>
      </Popover>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should render with default open styles', () => {
    const {
      baseElement,
    } = render(
      <Popover
        open
        anchorEl={document.createElement('div')}
      >
        <div>Inside</div>
      </Popover>,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
