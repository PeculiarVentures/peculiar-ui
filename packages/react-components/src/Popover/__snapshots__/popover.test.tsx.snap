// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover /> should render with default open styles 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 1300;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
}

.emotion-1 {
  background: var(--pv-color-black);
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
}

.emotion-2 {
  background: var(--pv-color-white);
  outline: 0;
  max-width: calc(100% - 32px);
  min-width: 16px;
  max-height: calc(100% - 32px);
  min-height: 16px;
  box-shadow: var(--pv-shadow-light-low);
  border-radius: 4px;
}

<body>
  <div />
  <div
    aria-hidden="false"
    class="emotion-0"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="emotion-1"
      style="opacity: 0.5; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      class="emotion-2"
      style="opacity: 1; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; position: absolute; left: 0px; top: 0px;"
      tabindex="-1"
    >
      <div>
        Inside
      </div>
    </div>
  </div>
</body>
`;

exports[`<Popover /> should render with default styles 1`] = `
<body>
  <div />
</body>
`;
